src/crypto/ms_oapxbc.rs

```diff
@@ -168,16 +208,35 @@ impl MsOapxbcSessionKey {
     }
 
     /// Use the session key to derive a one-time HMAC key for signing this JWS.
-    pub fn sign<V: JwsSignable>(&self, jws: &V) -> Result<V::Signed, JwtError> {
+    pub fn sign<T, V: JwsSignable>(
```

Comment:
First comment is that this change breaks signing without tpm. I've implemented my msal library to support either with or without tpm (this makes testing much easier). Would you wrap these tpm changes inside the hsm-crypto feature?
In msal, I've got different function definitions depending on whether the tpm feature is enabled. For example, this sign function could be changed to sign_internal (not pub) and wrap the common behavior, then have feature defined function headers for sign with/without tpm.

Reply:
It seems in hsm-crypto you provide either soft hsm, or tpm (if you enable the tpm feature).
I'd like to see the MS-OAPXBC stick around when hsm-crypto isn't enabled.